core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
As you might have noticed I removed the effects in https://github.com/suffuse/sfs/pull/30

> In what sense are you saying fuse is blocking?

All functions require you to return something. For example `getattr` requires you to tell it when a file does not exist. If this information came in through a `Future`, we would have to block on it. I'm unsure of the `read` semantics: "Returns the number of bytes transferred", will it try again if the `size` was larger than the amount of bytes transferred?